### ISSUE TITLE
Geoserver integration

### ISSUE THREAD

**Issue by geoHeil** [uber/kepler.gl#85]
are there plans to support geoserver out of the box as a data source?

**Comment by yiyaya66** [comment]
@geoHeil could you describe the use case here, share an example and the technology you are planning to use? If there is any documentation, it would be great if you could share with us.

**Comment by geoHeil** [comment]
Not sure what type of geospatial data storage you have at uber (would be
interested to learn more about that) in case of an open source setup
something like http://www.geomesa.org/ which acts as a geotools datastore
provider for geoserver would be my data provider.

Geoserver then publishes a feed - even real time - as wmf / wpf
standardized services.
I.e. It would be great if consumption of such services would work out of
the box.

Also an open source pendant of http://www.ccri.com/case-studies/stealth/
would be great, I.e simply adding CQL capabilities to a layer loaded from
geoserver should be mostly sufficient.
Ziyi Li <notifications@github.com> schrieb am Fr. 15. Juni 2018 um 20:41:

**Comment by ghirardinicola** [comment]
Looking for an integration with postgres/postgis but any server side aggregation would be fine. Any advice on how to develop as plugin or wrapper app would be useful

**Comment by macrigiuseppe** [comment]
We are gonna look into implementing data connectors along with examples to show their abilities

**Comment by heshan0131** [comment]
We can create an example with node, PostgreSQL with RESTful API. There is a (post)[https://blog.logrocket.com/setting-up-a-restful-api-with-node-js-and-postgresql-d96d6fc892d8/] can be referred to

**Comment by marcusnygren** [comment]
Is there any update on the use case that @geoHeil is proposing? :)

**Comment by mapsensedev** [comment]
Kepler should be standards compliant. From that perspective it should add support for all OGC standard data types like WMS, WFS, WMTS etc. What @geoHeil is asking is basically an implementation of those specifications. There are different servers like Geoserver, ESRI ArcGIS Server or Mapserver which implement OGC Map service specs

**Comment by michikommader** [comment]
I agree that having full OGC OWS compatibility would be nice. But we could start with a much simpler approach by adding a functionality under "Load Map using URL" to resolve RESTful HTTP queries which generate GeoJSON results, for example. There, you could simply paste a well-defined/standardized WFS GetFeature request which would generate a GeoJSON. This [dynamically created aeromap layer extract](https://ts.dlr.de/mumme-server/aeromap/ows?service=WFS&version=1.0.0&request=GetFeature&typeName=aeromap:roadmark&outputFormat=application/json&srsName=EPSG:4326&maxFeatures=150) could serve as an example. Right now I have to call this URL, download the returned result as `.json` and upload this file via "Load Files" to kepler.gl. This is a bit cumbersome.

Bonus: Exchanging `outputFormat=application/json` to `outputFormat=csv` in the above example URL will return a CSV with OGC WKT geometries which could be interesting for testing as well.

**Comment by DeepUpRosenthal** [comment]
Has this any progress?

**Comment by boonruang** [comment]
Has any solution?
